File: hrms-backend/src/main/java/sep490/com/example/hrms_backend/service/DepartmentService.java
package sep490.com.example.hrms_backend.service;

import sep490.com.example.hrms_backend.dto.DepartmentDTO;
import sep490.com.example.hrms_backend.dto.LineDTO;
import sep490.com.example.hrms_backend.dto.PositionDTO;

import java.util.List;

public interface DepartmentService {
    List<DepartmentDTO> getAllDepartments();
    List<PositionDTO> getPositionsByDepartment(Long departmentId);
    List<LineDTO> getLinesByDepartment(Long departmentId);
}

File: hrms-frontend/src/services/departmentService.js
import axiosClient from "./axiosClient";

const departmentService = {
  getAllDepartments: () => {
    return axiosClient.get("/departments");
  },
  getPositionsByDepartment: (departmentId) => {
    return axiosClient.get(`/departments/${departmentId}/positions`);
  },
  getLinesByDepartment: (departmentId) => {
    return axiosClient.get(`/departments/${departmentId}/lines`);
  },
};

export default departmentService;
